File: cpp_solutions/1404.cpp
class Solution {
public:
    int numSteps(string s) {
        int n = s.size();
        int carry = 0;
        int res = 0;
        for (int i = n - 1; i > 0; --i) {
            res += 1;
            if (s[i] - '0' + carry == 1) {
                carry = 1;
                res += 1;
            }
        }
        return res + carry;
    }
};

File: rust_solutions/1404.rs
impl Solution {
    pub fn num_steps(s: String) -> i32 {
        let n = s.len();
        let s_b = s.as_bytes();

        let mut carry = 0;
        let mut res = 0;
        
        for i in (1..n).rev() {
            res += 1;
            if s_b[i] - ('0' as u8) + (carry as u8) == 1 {
                carry = 1;
                res += 1;
            }
        }
        return res + carry;
    }
}
